modules/contentpool_remote_register/src/Plugin/rest/resource/ContentpoolTermReferenceFieldsResource.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+namespace Drupal\contentpool_remote_register\Plugin\rest\resource;
+
+use Drupal\Core\Entity\EntityFieldManagerInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\rest\ModifiedResourceResponse;
+use Drupal\rest\Plugin\ResourceBase;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * A rest resource that provides term reference fields for satellites.
+ *
+ * @RestResource(
+ *   id = "contentpool:contentpool_term_reference_fields",
+ *   label = "Contentpool term reference fields",
+ *   uri_paths = {
+ *     "canonical" = "/api/contentpool-term-reference-fields",
```

Comment:
I think you need to grant the permission to the replicator role, right? Please so in contentpool_install() then.